### PR TITLE
Add Puppet8x support

### PR DESCRIPTION
**Pull request by flepoutre**

### CODE REVIEW

File: manifests/alias.pp [hunk 1]
```diff
@@ -1,40 +1,49 @@
...
+  Optional[String] $uri                                    = undef,
+  Optional[String] $db_url                                 = undef,
+  Optional[Variant[Hash, String]] $path_aliases            = undef,
+  Optional[String] $ssh_options                            = undef,
+  Optional[String] $remote_host                            = undef,
+  Optional[String] $remote_user                            = undef,
+  Optional[String] $custom_options                         = undef,
+  Optional[Variant[Hash, String]] $command_specific        = undef,
+  Optional[Variant[Hash, String]] $source_command_specific = undef,
+  Optional[Variant[Hash, String]] $target_command_specific = undef,
```
Comment (jonhattan) [inline manifests/alias.pp:34]: I don't understand why you're changing the types of several params in this define.

File: manifests/alias.pp [hunk 2]
```diff
@@ -1,40 +1,49 @@
...
-
   if (!defined(Class['drush'])) {
     fail('You must include class drush before declaring aliases')
   }
 
-  if $parent {
-    if $parent !~ /^@/ {
-      fail("Invalid parent alias '${parent}'. Parent aliases must start with @.")
-    }
-  }
```
Comment (jonhattan) [inline manifests/alias.pp:29]: This change is not desired. It has nothing to do with _puppet 8x support_

File: manifests/aliasng.pp [hunk 3]
```diff
@@ -1,23 +1,38 @@
...
+  Optional[String] $uri                                    = undef,
+  Optional[String] $db_url                                 = undef,
+  Optional[Variant[Hash, String]] $path_aliases            = undef,
+  Optional[String] $ssh_options                            = undef,
+  Optional[String] $remote_host                            = undef,
+  Optional[String] $remote_user                            = undef,
+  Optional[String] $custom_options                         = undef,
+  Optional[Variant[Hash, String]] $command_specific        = undef,
+  Optional[Variant[Hash, String]] $source_command_specific = undef,
+  Optional[Variant[Hash, String]] $target_command_specific = undef,
```
Comment (jonhattan) [inline manifests/aliasng.pp:34]: Same than above, I don't understand why you're changing the types of several params in this define.

File: manifests/config.pp [hunk 4]
```diff
@@ -32,19 +31,15 @@
           alias-path => [
             '/etc/drush/sites',
           ],
-          include => [
-            '/usr/share/drush/commands',
-          ],
```
Comment (jonhattan) [inline manifests/config.pp:37]: ?

File: manifests/init.pp [hunk 5]
```diff
@@ -51,22 +51,21 @@
...
-  Boolean $bash_autocompletion                   = true,
-  Array $extensions                              = [],
-  Hash $aliases                                  = {},
-  Stdlib::Absolutepath $composer_path            = '/usr/local/bin/composer',
-  Optional[Stdlib::Absolutepath] $php_path       = undef,
-  Optional[Stdlib::Absolutepath] $php_ini_path   = undef,
-  Optional[Stdlib::Absolutepath] $drush_ini_path = undef,
+class drush (
+  Array[String] $versions          = ['7',],
+  String $default_version          = '7',
```
Comment (jonhattan) [inline manifests/init.pp:56]: Why do you change the default value of these params?

File: manifests/init.pp [hunk 6]
```diff
@@ -51,22 +51,21 @@
...
+  Boolean $ensure_extra_packages   = false,
+  Array $extra_packages            = $drush::params::extra_packages,
+  Boolean $bash_integration        = false,
+  Boolean $bash_autocompletion     = true,
+  Array $extensions                = [],
+  Hash $aliases                    = {},
+  String $composer_path            = '/usr/local/bin/composer',
+  Optional[String] $php_path       = undef,
+  Optional[String] $php_ini_path   = undef,
+  Optional[String] $drush_ini_path = undef,
```
Comment (jonhattan) [inline manifests/init.pp:67]: Same as before. There're wrong type changes in this class.

File: manifests/setup.pp [hunk 7]
```diff
@@ -54,14 +53,12 @@
   }
 
   # Drush directories.
-  file { ['/etc/drush', '/usr/share/drush', '/usr/share/drush/commands', '/usr/share/drush/commands/Commands']:
+  file { ['/etc/drush', '/usr/share/drush', '/usr/share/drush/commands']:
```
Comment (jonhattan) [inline manifests/setup.pp:56]: ?

File: manifests/setup.pp [hunk 8]
```diff
@@ -54,14 +53,12 @@
...
     ensure => directory,
   }
   if $drush::modern {
     file { '/etc/drush/sites':
-      ensure  => directory,
-      purge   => true,
-      recurse => true,
+      ensure => directory,
```
Comment (jonhattan) [inline manifests/setup.pp:61]: ?

**Comment by flepoutre** [comment]
Hi,
Sorry I left from the last tag 0.6.1 and not master branch ...

**Comment by flepoutre** [comment]
Ok is it better like this ?
Thanks.

**Comment by jonhattan** [comment]
Thanks!

Squashed and merged.

I'll do a release soon with this and other changes.

**Comment by flepoutre** [comment]
@jonhattan Hello, any news on "I'll do a release soon with this and other changes." ?

**Comment by jonhattan** [comment]
Sorry for the delay. Just released https://forge.puppet.com/modules/jonhattan/drush/readme

**Pull request merged**